src/services/pokemon.js

```diff
@@ -0,0 +1,102 @@
+import { pokemonMunger } from "../utils/helper";
+
+
+export const fetchPokemon = async () => {
+    const res = await fetch(`https://pokedex-alchemy.herokuapp.com/api/pokedex?&perPage=20`);
+    const pokemonData = await res.json();
+    const pokemonMap = await pokemonData.results.map((item) => 
+    pokemonMunger(item));
+
+
+    return pokemonMap
+}
+
+// export const fetchPokemonPage = async (page) => {
+//     const res = await fetch(`https://pokedex-alchemy.herokuapp.com/api/pokedex?page=${page}&perPage=21`);
+//     const pokemonData = await res.json();
+//     const pokemonMap = await pokemonData.results.map((item) => 
+//     pokemonMunger(item));
+
+
+//     return pokemonMap
+// }
+
+export const fetchTypes = async () => {
+    const res = await fetch('https://pokedex-alchemy.herokuapp.com/api/pokedex/types');
+    const typeData = await res.json();
+    const typeMap = await typeData.map((item) =>
+     ({type: item.type}));
+
+     return typeMap;
+}
+
+export const fetchFilterTypes = async (page, type) => {
+    const res = await fetch(`https://pokedex-alchemy.herokuapp.com/api/pokedex?perPage=20&type=${type}`);
+    const filterData = await res.json();
+    const filterMap = filterData.results.map((item) =>

```

Comment:
Remember that since you only care about the `results` array on the returned object, you can simply destructure it from the `res.json()` call directly and slim up the code some. Example: 

```js
const { results } = await res.json();
return results.map((item) => pokemonMunger(item));
```